### PR TITLE
feat(clerk-js,types): Build Plan & Billing page

### PR DESCRIPTION
**Pull request by nikospapcom**
## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/common/CurrentPlanSection.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import { Box, localizationKeys, Text, useLocalizations } from '../customizables';
+import { ProfileSection } from '../elements';
+import { mqu } from '../styledSystem';
+import { centsToUnit, getRelativeToNowDateKey } from '../utils';
+
+export const CurrentPlanSection = ({ currentPlan }: { currentPlan?: BillingPlanResource | null }) => {
+  const { t } = useLocalizations();
+
+  if (!currentPlan) {
```
Comment (AlexNti) [inline packages/clerk-js/src/ui/common/CurrentPlanSection.tsx:11]: We can omit this assertion, current plan will always be initiated after the enabling of billing

File: packages/clerk-js/src/ui/common/CurrentPlanSection.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      <ProfileSection.Item id='currentPlan'>
+        <Box>
+          <Text
+            variant='subtitle'
+            sx={t => ({ marginBottom: t.space.$2 })}
+          >
+            {currentPlan.name}
+          </Text>
+          <Text sx={t => ({ fontWeight: t.fontWeights.$semibold, fontSize: t.fontSizes.$lg })}>
+            {centsToUnit(currentPlan.priceInCents)}{' '}
```
Comment (AlexNti) [inline packages/clerk-js/src/ui/common/CurrentPlanSection.tsx:31]: We should use the same function to `OrganizationPlan` as well.

File: packages/clerk-js/src/ui/components/OrganizationProfile/OrganizationBillingPage.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { useOrganization } from '@clerk/shared/react';
+
+import { CurrentPlanSection, PaymentMethodSection } from '../../common';
+import { Button, Col, descriptors, Flex, localizationKeys, useLocalizations } from '../../customizables';
+import { FullHeightLoader, Header, withCardStateProvider } from '../../elements';
+import { useFetch } from '../../hooks';
+
+export const OrganizationBillingPage = withCardStateProvider(() => {
```
Comment (AlexNti) [inline packages/clerk-js/src/ui/components/OrganizationProfile/OrganizationBillingPage.tsx:8]: Do we need `withCardStateProvider` ?

File: packages/localizations/src/en-US.ts [hunk 4]
```diff
@@ -802,6 +802,19 @@ export const enUS: LocalizationResource = {
     },
   },
   billing: {
+    start: {
+      headerTitle__billing: 'Plan & Billing',
+    },
+    manageBillingInfo: 'Manage billing info',
```
Comment (AlexNti) [inline packages/localizations/src/en-US.ts:808]: How about create a parent with a key that will reference the name of the page simillar to what we are doing to `managePlanScreen` ?

**Review by AlexNti — APPROVED** [review]
💯

File: packages/clerk-js/src/ui/components/OrganizationProfile/OrganizationBillingPage.tsx (hunk 3 above)
Comment (nikospapcom) [reply packages/clerk-js/src/ui/components/OrganizationProfile/OrganizationBillingPage.tsx:10]: Yes we need in order to get card state

File: packages/localizations/src/en-US.ts (hunk 4 above)
Comment (nikospapcom) [reply packages/localizations/src/en-US.ts]: Done

File: packages/clerk-js/src/ui/common/CurrentPlanSection.tsx (hunk 2 above)
Comment (nikospapcom) [reply packages/clerk-js/src/ui/common/CurrentPlanSection.tsx:31]: yes, lets address in a following pr

File: packages/clerk-js/src/ui/common/CurrentPlanSection.tsx (hunk 1 above)
Comment (nikospapcom) [reply packages/clerk-js/src/ui/common/CurrentPlanSection.tsx:11]: as we discussed offline we are keeping for now

**Review by panteliselef — APPROVED** [review]

File: packages/clerk-js/src/ui/common/CurrentPlanSection.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import type { BillingPlanResource } from '@clerk/types';
+
+import { Box, localizationKeys, Text, useLocalizations } from '../customizables';
+import { ProfileSection } from '../elements';
+import { mqu } from '../styledSystem';
+import { centsToUnit, getRelativeToNowDateKey } from '../utils';
+
+export const CurrentPlanSection = ({ currentPlan }: { currentPlan?: BillingPlanResource | null }) => {
```
[the remark this answers is not in the document]
Comment (nikospapcom) [reply packages/clerk-js/src/ui/common/CurrentPlanSection.tsx:8]: @panteliselef I'm going to raise a new pr in order to not block @AlexNti

File: packages/clerk-js/src/ui/common/PaymentMethodSection.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      centered={false}
+      sx={{ [mqu.md]: { alignItems: 'flex-start' } }}
+    >
+      <ProfileSection.Item id='paymentMethod'>
+        {currentPlan.paymentMethod ? (
+          <Flex sx={{ flexDirection: 'column' }}>
+            <Text>•••• {currentPlan.paymentMethod.card.last4}</Text>
+            <Text sx={t => ({ color: t.colors.$colorTextSecondary, fontSize: t.fontSizes.$sm })}>
+              {t(localizationKeys('billing.paymentMethodSection.expires'))} {currentPlan.paymentMethod.card.expMonth}/
+              {currentPlan.paymentMethod.card.expYear}
```
[the remark this answers is not in the document]
Comment (nikospapcom) [reply packages/clerk-js/src/ui/common/PaymentMethodSection.tsx]: @panteliselef I'm going to raise a new pr in order to not block @AlexNti

**Pull request merged**